extensions/validation/src/main/java/org/mule/extension/validation/internal/CommonValidationOperations.java

```diff
@@ -0,0 +1,519 @@
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.validation.internal;
+
+import static org.mule.extension.validation.internal.ImmutableValidationResult.error;
+import static org.mule.extension.validation.internal.ImmutableValidationResult.ok;
+import org.mule.api.MuleEvent;
+import org.mule.api.NestedProcessor;
+import org.mule.extension.annotations.ImplementationOf;
+import org.mule.extension.annotations.Operation;
+import org.mule.extension.annotations.ParameterGroup;
+import org.mule.extension.annotations.RestrictedTo;
+import org.mule.extension.annotations.param.Optional;
+import org.mule.extension.validation.ValidationExtension;
+import org.mule.extension.validation.api.ValidationException;
+import org.mule.extension.validation.api.ValidationResult;
+import org.mule.extension.validation.api.Validator;
+import org.mule.extension.validation.validator.BooleanValidator;
+import org.mule.extension.validation.validator.CreditCardNumberValidator;
+import org.mule.extension.validation.validator.CreditCardType;
+import org.mule.extension.validation.validator.DomainCountryCodeValidator;
+import org.mule.extension.validation.validator.DomainValidator;
+import org.mule.extension.validation.validator.DoubleValidator;
+import org.mule.extension.validation.validator.EmailValidator;
+import org.mule.extension.validation.validator.EmptyValidator;
+import org.mule.extension.validation.validator.FloatValidator;
+import org.mule.extension.validation.validator.ISBN10Validator;
+import org.mule.extension.validation.validator.ISBN13Validator;
+import org.mule.extension.validation.validator.IntegerValidator;
+import org.mule.extension.validation.validator.IpValidator;
+import org.mule.extension.validation.validator.LongValidator;
+import org.mule.extension.validation.validator.MatchesRegexValidator;
+import org.mule.extension.validation.validator.NotEmptyValidator;
+import org.mule.extension.validation.validator.NotNullValidator;
+import org.mule.extension.validation.validator.NullValidator;
+import org.mule.extension.validation.validator.NumberValidationOptions;
+import org.mule.extension.validation.validator.ShortValidator;
+import org.mule.extension.validation.validator.SizeValidator;
+import org.mule.extension.validation.validator.TimeValidator;
+import org.mule.extension.validation.validator.TopLevelDomainValidator;
+import org.mule.extension.validation.validator.UrlValidator;
+import org.mule.transport.NullPayload;
+
+import com.google.common.base.Joiner;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang.exception.ExceptionUtils;
+
+/**
+ * Defines the operations of {@link ValidationExtension}
+ * which executes the {@link Validator}s that the extension
+ * provides out of the box
+ *
+ * @see ValidationExtension
+ * @since 3.7.0
+ */
+@ImplementationOf(ValidationExtension.class)
```

Comment:
We discussed this personally. This is not a two way reference but an improvement will be made in the ext-api so that operations of an extensible extension do not need to explicitelly set this. That will be however a different issue